blivet/blivet.py

```diff
@@ -791,7 +791,12 @@ def create_device(self, device):
         is_snapshot = isinstance(device, LVMLogicalVolumeDevice) and device.is_snapshot_lv
 
         if device.format.type and not device.format_immutable and not is_snapshot:
-            self.devicetree.actions.add(ActionCreateFormat(device))
+            try:
+                self.devicetree.actions.add(ActionCreateFormat(device))
+            except:  # pylint: disable=broad-except
+                # revert devicetree changes no matter what happened
+                self.reset_device(device)
```

Comment:
This will cancel all actions scheduled for the device, which may not be warranted. A slightly less heavy-handed approach would be to remove only the actions you create in this method on failure. You would need to do a few things to make it work nicely, though:

1. change `DeviceAction.__init__` to raise `blivet.errors.DependencyError` instead of `ValueError`, then catch only that exception in `create_device`
2. be prepared for an exception to be raised late in `remove`, which you can safely ignore